src/scaffoldmaker/meshtypes/meshtype_3d_stellate1.py

```diff
@@ -0,0 +1,543 @@
+                        elementIdentifier += 1
+        fm.endChange()
+        return []
+
+    @classmethod
+    def refineMesh(cls, meshrefinement, options):
+        """
+        Refine source mesh into separate region, with change of basis.
+        :param meshrefinement: MeshRefinement, which knows source and target region.
+        :param options: Dict containing options. See getDefaultOptions().
+        """
+        assert isinstance(meshrefinement, MeshRefinement)
+        refineElementsCount1 = options['Refine number of elements along arm']
+        refineElementsCount2 = options['Refine number of elements across arm']
+        refineElementsCount3 = options['Refine number of elements through thickness']
+        meshrefinement.refineAllElementsCubeStandard3d(refineElementsCount1, refineElementsCount2, refineElementsCount3)
+
+
+def createArm(thAr, elementLengths, armAngle, armAngleConst, elementsCount, dipMultiplier, numArm, na):
+    """
+    Create single arm unit.
+    Base length of element is 1.
+    Direction: anticlockwise.
+    Minimum arm length is 2 elements.
+    :param thAr: angle arising from base node (rad)
+    :param elementLengths: [Element length along arm, half Element width across arm, Element thickness]
+    :param armAngle: angle from x axis of the arm about origin (rad)
+    :param armAngleConst: minimum angle from x axis of the first arm about origin (rad)
+    :param elementsCount: element counts in x,y,z directions
+    :param dipMultiplier: factor that wheel nodes protrude by, relative to unit length
+    :param numArm: number of arms in body
+    :param na: nth arm currently being created
+    :return: x: positions of nodes in arm
+    :return: nodeIdentifier: number of last node in arm +1
+    :return: xnodes_ds1: ds1 derivatives of nodes associated with x
+    :return: xnodes_ds2: ds2 derivatives of nodes associated with y
+    """
+
+    elementsCount1, elementsCount2, elementsCount3 = elementsCount
+    [elen, ewid, zheight] = elementLengths
+
+    xnodes_ds1 = []
+    xnodes_ds2 = []
+    dx_ds1 = [elen, 0.0, 0.0]
+    dx_ds2 = [0.0, ewid, 0.0]
+    wheelDvMult = [0.5, 0.75]
+
+    nodes_per_layer = (elementsCount1 + 1) * (elementsCount2 + 1) - 2 # discount 2 armEnd corners
+    x = []
+
+    # nCentre and rmVertexNodes are in pythonic indexing
+    nCentre = elementsCount1
+    nCentre = [nCentre, nCentre + nodes_per_layer]
+    if na == 0:
+        rmVertexNodes = []
+    elif na != numArm - 1:
+        rmVertexNodes = nCentre + [0, nodes_per_layer]
+    else:
+        rmVertexNodes = nCentre + [0, nodes_per_layer,
+                          2* (elementsCount1+1) - 1, 2* (elementsCount1+1) - 1 + nodes_per_layer ]
+    dcent = [elen * math.cos(armAngleConst / 2), elen * math.sin(armAngleConst / 2), 0.0]
+    dvertex = [round(elen * dipMultiplier * math.cos(thAr), 12),
+                round(elen * dipMultiplier * math.sin(thAr), 12)]
```

Comment:
This should be mean of element length and width by default.
```
    dipLength = 0.5*(elen + ewid)*dipMultiplier
    dvertex = [round(dipLength * math.cos(thAr), 12),
               round(dipLength * math.sin(thAr), 12)]
    dipMag = 2*dipLength - elen
```
Magnitude of d2 on outside should later be set to dipMag.
Why are these rounded? Avoid?